components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java

```diff
@@ -0,0 +1,291 @@
+     * Validate the secret post request.
+     *
+     * @param secretAddRequest secret post request.
+     */
+    private void validateSecretAddRequest(SecretAddRequest secretAddRequest) {
+
+        String secretAddName = secretAddRequest.getName();
+        if (StringUtils.isBlank(secretAddName)) {
+            throw handleException(Response.Status.BAD_REQUEST, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_REFERENCE_NAME_NOT_SPECIFIED, null);
+        }
+        if (StringUtils.isBlank(secretAddRequest.getValue())) {
+            throw handleException(Response.Status.BAD_REQUEST, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_SECRET_VALUE_NOT_SPECIFIED, null);
+        }
+    }
+
+    /**
+     * Build secret requestDTO by secret body request.
+     *
+     * @param secretAddRequest secret post body.
+     * @return Secret requestDTO object.
+     */
+    private Secret buildSecretRequestDTOFromSecretAddRequest(SecretAddRequest secretAddRequest) {
+
+        Secret requestDTO = new Secret();
+        requestDTO.setSecretName(secretAddRequest.getName());
+        requestDTO.setSecretValue(secretAddRequest.getValue());
+        requestDTO.setDescription(secretAddRequest.getDescription());
+        return requestDTO;
+    }
+
+    /**
+     * Delete a secret sender by name.
+     *
+     * @param name Name of the secret.
+     */
+    public void deleteSecret(String secretType, String name) {
+
+        try {
+            SecretManagementServiceHolder.getSecretConfigManager().deleteSecret(secretType, name);
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_DELETING_SECRET, name);
+        }
+    }
+
+    /**
+     * Retrieve the secret details by name.
+     *
+     * @param name secret name.
+     * @return secret.
+     */
+    public SecretResponse getSecret(String secretType, String name) {
+
+        try {
+            Secret responseDTO = SecretManagementServiceHolder.getSecretConfigManager().getSecret(secretType, name);
+            SecretResponse secretResponse = new SecretResponse();
+            secretResponse.secretName(responseDTO.getSecretName());
+            secretResponse.setCreated(responseDTO.getCreatedTime());
+            secretResponse.setLastModified(responseDTO.getLastModified());
+            secretResponse.setSecretId(responseDTO.getSecretId());
+            secretResponse.setType(responseDTO.getSecretType());
+            secretResponse.setDescription(responseDTO.getDescription());
+            return secretResponse;
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_GETTING_SECRET, name);
+        }
+    }
+
+    /**
+     * Retrieve all the secrets of the tenant.
+     *
+     * @return secrets of the tenant.
+     */
+
```

Comment:
Let's remove this new line.

Reply:
Addressed with commit [a6f4632](https://github.com/wso2/identity-api-server/pull/315/commits/a6f463218febb07c19f260f0aa26e33fdec47de3)